default/tests/shared/database/source_control/test_working_dir.py

```diff
@@ -66,32 +68,102 @@ def test_list(self):
             'access_type': 'restricted'
         }, self.session)
 
+        mock_perm_result = Mock()
+        mock_perm_result.allow_all = False
+        mock_perm_result.allowed_object_id_list = [ds1.id, ds2.id, ds3.id]
+        mock_get_dataset_viewing_permissions.return_value = mock_perm_result
```

Comment:
`list` now depends on `get_dataset_viewing_permissions`, so we can just mock out that method rather than using real RoleMembers